lib/Driver/Compilation.cpp

```diff
@@ -313,6 +320,20 @@ namespace driver {
       }
     }
 
+    void
+    forEachPidOrQuasiPid(const Job *MaybeBatchJob, ProcessId Pid,
+                         llvm::function_ref<void(const Job *, int64_t)> fn) const {
+      if (BatchJobs.count(MaybeBatchJob) != 0) {
```

Comment:
I had to stare at this and check out `BatchJobs` and think to verify that this line was correctly determining if `MaybeBatchJob` was a batch job. My style would be to add a one-line predicate function called `isBatchJob` may with a comment saying "all batch jobs must be in BatchJobs", somewhere. Or just adding that comment here could work.

Reply:
Factored out `isBatchJob` though it only has one user now. Minor improvements -- why not?